## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [1]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [2]:
# Write your code for importing the csv file here
movies = pd.read_csv("Movies.csv")
movies

color      director_name  num_critic_for_reviews  duration   
0     Color      James Cameron                   723.0     178.0  \
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name   
0                         0.0                   855.0    Joel David Moore  \
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross   
0                     1000.0  760505847.0  \
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language   
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English  \
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year   
0             USA           PG-13  237000000.0      2009.0  \
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [3]:
# Write your code for inspection here
movies.shape
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      3851 non-null   object 
 1   director_name              3853 non-null   object 
 2   num_critic_for_reviews     3852 non-null   float64
 3   duration                   3852 non-null   float64
 4   director_facebook_likes    3853 non-null   float64
 5   actor_3_facebook_likes     3847 non-null   float64
 6   actor_2_name               3852 non-null   object 
 7   actor_1_facebook_likes     3853 non-null   float64
 8   gross                      3853 non-null   float64
 9   genres                     3853 non-null   object 
 10  actor_1_name               3853 non-null   object 
 11  movie_title                3853 non-null   object 
 12  num_voted_users            3853 non-null   int64  
 13  cast_total_facebook_likes  3853 non-null   int64

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='green'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [4]:
(movies.isnull().sum()>0).sum()

12

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='green'>12</font>

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [5]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations
movies.drop(["color",
"director_facebook_likes",
"actor_1_facebook_likes",
"actor_2_facebook_likes",
"actor_3_facebook_likes",
"actor_2_name",
"cast_total_facebook_likes",
"actor_3_name",
"duration",
"facenumber_in_poster",
"content_rating",
"country",
"movie_imdb_link",
"aspect_ratio",
"plot_keywords"], axis=1,inplace=True)

movies


director_name  num_critic_for_reviews        gross   
0         James Cameron                   723.0  760505847.0  \
1        Gore Verbinski                   302.0  309404152.0   
2            Sam Mendes                   602.0  200074175.0   
3     Christopher Nolan                   813.0  448130642.0   
4        Andrew Stanton                   462.0   73058679.0   
...                 ...                     ...          ...   
3848      Shane Carruth                   143.0     424760.0   
3849   Neill Dela Llana                    35.0      70071.0   
3850   Robert Rodriguez                    56.0    2040920.0   
3851       Edward Burns                    14.0       4584.0   
3852           Jon Gunn                    43.0      85222.0   

                                   genres     actor_1_name   
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder  \
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users   
0                                       Avatar            886204  \
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language       budget  title_year  imdb_score   
0                   3054.0  English  237000000.0      2009.0         7.9  \
1                   1238.0  English  300000000.0      2007.0         7.1   
2                    994.0  English  245000000.0      2015.0         6.8   
3                   2701.0  English  250000000.0      2012.0         8.5   
4                    738.0  English  263700000.0      2012.0         6.6   
...                    ...      ...          ...         ...         ...   
3848                 371.0  English       7000.0      2004.0         7.0   
3849                  35.0  English       7000.0      2005.0         6.3   
3850                 130.0  Spanish       7000.0      1992.0         6.9   
3851                  14.0  English       9000.0      2011.0         6.4   
3852                  84.0  English       1100.0      2004.0         6.6   

      movie_facebook_likes  
0                    33000  
1                        0  
2                    85000  
3                   164000  
4                    24000  
...                    ...  
3848                 19000  
3849                    74  
3850                     0  
3851                   413  
3852                   456  

[3853 rows x 13 columns]

In [6]:
movies.columns.size

13

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='green'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'.

In [8]:
# Write you code here
percentage_null=movies.isnull().sum()*100/len(movies)
percentage_null

director_name             0.000000
num_critic_for_reviews    0.025954
gross                     0.000000
genres                    0.000000
actor_1_name              0.000000
movie_title               0.000000
num_voted_users           0.000000
num_user_for_reviews      0.000000
language                  0.103815
budget                    0.000000
title_year                0.000000
imdb_score                0.000000
movie_facebook_likes      0.000000
dtype: float64

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='green'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [ ]:
# Write your code for filling the NaN values in the 'language' column here
movies.language.fillna("English",inplace=True)

In [ ]:
movies["language"].value_counts()

English       3674
French          37
Spanish         26
Mandarin        14
German          13
Japanese        12
Hindi           10
Cantonese        8
Italian          7
Portuguese       5
Korean           5
Norwegian        4
Dutch            3
Persian          3
Thai             3
Danish           3
Aboriginal       2
Dari             2
Indonesian       2
Hebrew           2
Russian          1
Romanian         1
Vietnamese       1
Arabic           1
Dzongkha         1
Zulu             1
None             1
Aramaic          1
Czech            1
Telugu           1
Icelandic        1
Filipino         1
Hungarian        1
Maya             1
Bosnian          1
Mongolian        1
Kazakh           1
Swedish          1
Name: language, dtype: int64

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='green'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [ ]:
# Write your code for unit conversion here
movies["budget"]=(movies.budget/1000000).round(2)
movies["gross"]= (movies.gross/1000000).round(2)
movies

director_name  num_critic_for_reviews   gross  \
0         James Cameron                   723.0  760.51   
1        Gore Verbinski                   302.0  309.40   
2            Sam Mendes                   602.0  200.07   
3     Christopher Nolan                   813.0  448.13   
4        Andrew Stanton                   462.0   73.06   
...                 ...                     ...     ...   
3848      Shane Carruth                   143.0    0.42   
3849   Neill Dela Llana                    35.0    0.07   
3850   Robert Rodriguez                    56.0    2.04   
3851       Edward Burns                    14.0    0.00   
3852           Jon Gunn                    43.0    0.09   

                                   genres     actor_1_name  \
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users  \
0                                       Avatar            886204   
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language  budget  title_year  imdb_score  \
0                   3054.0  English  237.00      2009.0         7.9   
1                   1238.0  English  300.00      2007.0         7.1   
2                    994.0  English  245.00      2015.0         6.8   
3                   2701.0  English  250.00      2012.0         8.5   
4                    738.0  English  263.70      2012.0         6.6   
...                    ...      ...     ...         ...         ...   
3848                 371.0  English    0.01      2004.0         7.0   
3849                  35.0  English    0.01      2005.0         6.3   
3850                 130.0  Spanish    0.01      1992.0         6.9   
3851                  14.0  English    0.01      2011.0         6.4   
3852                  84.0  English    0.00      2004.0         6.6   

      movie_facebook_likes  
0                    33000  
1                        0  
2                    85000  
3                   164000  
4                    24000  
...                    ...  
3848                 19000  
3849                    74  
3850                     0  
3851                   413  
3852                   456  

[3853 rows x 13 columns]

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [ ]:
# Write your code for creating the profit column here
movies["profit"]=movies["gross"]-movies["budget"]
movies

director_name  num_critic_for_reviews   gross  \
0         James Cameron                   723.0  760.51   
1        Gore Verbinski                   302.0  309.40   
2            Sam Mendes                   602.0  200.07   
3     Christopher Nolan                   813.0  448.13   
4        Andrew Stanton                   462.0   73.06   
...                 ...                     ...     ...   
3848      Shane Carruth                   143.0    0.42   
3849   Neill Dela Llana                    35.0    0.07   
3850   Robert Rodriguez                    56.0    2.04   
3851       Edward Burns                    14.0    0.00   
3852           Jon Gunn                    43.0    0.09   

                                   genres     actor_1_name  \
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users  \
0                                       Avatar            886204   
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language  budget  title_year  imdb_score  \
0                   3054.0  English  237.00      2009.0         7.9   
1                   1238.0  English  300.00      2007.0         7.1   
2                    994.0  English  245.00      2015.0         6.8   
3                   2701.0  English  250.00      2012.0         8.5   
4                    738.0  English  263.70      2012.0         6.6   
...                    ...      ...     ...         ...         ...   
3848                 371.0  English    0.01      2004.0         7.0   
3849                  35.0  English    0.01      2005.0         6.3   
3850                 130.0  Spanish    0.01      1992.0         6.9   
3851                  14.0  English    0.01      2011.0         6.4   
3852                  84.0  English    0.00      2004.0         6.6   

      movie_facebook_likes  profit  
0                    33000  523.51  
1                        0    9.40  
2                    85000  -44.93  
3                   164000  198.13  
4                    24000 -190.64  
...                    ...     ...  
3848                 19000    0.41  
3849                    74    0.06  
3850                     0    2.03  
3851                   413   -0.01  
3852                   456    0.09  

[3853 rows x 14 columns]

In [ ]:
# Write your code for sorting the dataframe here
sorted = movies.sort_values("profit",ascending=False,ignore_index=True)

In [ ]:
top10 = sorted.head(10)

In [ ]:
top10

director_name  num_critic_for_reviews   gross  \
0      James Cameron                   723.0  760.51   
1    Colin Trevorrow                   644.0  652.18   
2      James Cameron                   315.0  658.67   
3       George Lucas                   282.0  460.94   
4   Steven Spielberg                   215.0  434.95   
5        Joss Whedon                   703.0  623.28   
6       Roger Allers                   186.0  422.78   
7       George Lucas                   320.0  474.54   
8  Christopher Nolan                   645.0  533.32   
9          Gary Ross                   673.0  408.00   

                                     genres         actor_1_name  \
0           Action|Adventure|Fantasy|Sci-Fi          CCH Pounder   
1          Action|Adventure|Sci-Fi|Thriller  Bryce Dallas Howard   
2                             Drama|Romance    Leonardo DiCaprio   
3           Action|Adventure|Fantasy|Sci-Fi        Harrison Ford   
4                             Family|Sci-Fi         Henry Thomas   
5                   Action|Adventure|Sci-Fi      Chris Hemsworth   
6  Adventure|Animation|Drama|Family|Musical    Matthew Broderick   
7           Action|Adventure|Fantasy|Sci-Fi      Natalie Portman   
8               Action|Crime|Drama|Thriller       Christian Bale   
9           Adventure|Drama|Sci-Fi|Thriller    Jennifer Lawrence   

                                  movie_title  num_voted_users  \
0                                     Avatar            886204   
1                             Jurassic World            418214   
2                                    Titanic            793059   
3         Star Wars: Episode IV - A New Hope            911097   
4                 E.T. the Extra-Terrestrial            281842   
5                               The Avengers            995415   
6                              The Lion King            644348   
7  Star Wars: Episode I - The Phantom Menace            534658   
8                            The Dark Knight           1676169   
9                           The Hunger Games            701607   

   num_user_for_reviews language  budget  title_year  imdb_score  \
0                3054.0  English   237.0      2009.0         7.9   
1                1290.0  English   150.0      2015.0         7.0   
2                2528.0  English   200.0      1997.0         7.7   
3                1470.0  English    11.0      1977.0         8.7   
4                 515.0  English    10.5      1982.0         7.9   
5                1722.0  English   220.0      2012.0         8.1   
6                 656.0  English    45.0      1994.0         8.5   
7                3597.0  English   115.0      1999.0         6.5   
8                4667.0  English   185.0      2008.0         9.0   
9                1959.0  English    78.0      2012.0         7.3   

   movie_facebook_likes  profit  
0                 33000  523.51  
1                150000  502.18  
2                 26000  458.67  
3                 33000  449.94  
4                 34000  424.45  
5                123000  403.28  
6                 17000  377.78  
7                 13000  359.54  
8                 37000  348.32  
9                140000  330.00

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

In [ ]:
top10.iloc[4]

director_name                        Steven Spielberg
num_critic_for_reviews                          215.0
gross                                          434.95
genres                                  Family|Sci-Fi
actor_1_name                             Henry Thomas
movie_title               E.T. the Extra-Terrestrial 
num_voted_users                                281842
num_user_for_reviews                            515.0
language                                      English
budget                                           10.5
title_year                                     1982.0
imdb_score                                        7.9
movie_facebook_likes                            34000
profit                                         424.45
Name: 4, dtype: object

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='green'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000.

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [ ]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe
# and name that dataframe as 'IMDb_Top_250'
top250=top[(movies.sort_values("imdb_score",ascending=False))["num_voted_users"]>25000].head(250)
top250["Rank"]=np.arange(1,251)
top250

director_name  num_critic_for_reviews   gross  \
1795        Frank Darabont                   199.0   28.34   
3016  Francis Ford Coppola                   208.0  134.82   
2543  Francis Ford Coppola                   149.0   57.30   
64       Christopher Nolan                   645.0  533.32   
325          Peter Jackson                   328.0  377.02   
...                    ...                     ...     ...   
2708      David O. Russell                   410.0   93.57   
22           Peter Jackson                   509.0  258.36   
1612         James Mangold                   291.0  119.52   
3237          Duncan Jones                   415.0    5.01   
874         Andrew Adamson                   212.0  267.65   

                                         genres           actor_1_name  \
1795                                Crime|Drama         Morgan Freeman   
3016                                Crime|Drama              Al Pacino   
2543                                Crime|Drama         Robert De Niro   
64                  Action|Crime|Drama|Thriller         Christian Bale   
325              Action|Adventure|Drama|Fantasy          Orlando Bloom   
...                                         ...                    ...   
2708                      Biography|Drama|Sport         Christian Bale   
22                            Adventure|Fantasy           Aidan Turner   
1612              Biography|Drama|Music|Romance  Sandra Ellis Lafferty   
3237                       Drama|Mystery|Sci-Fi           Kevin Spacey   
874   Adventure|Animation|Comedy|Family|Fantasy       Kathleen Freeman   

                                         movie_title  num_voted_users  \
1795                       The Shawshank Redemption           1689764   
3016                                  The Godfather           1155770   
2543                         The Godfather: Part II            790926   
64                                  The Dark Knight           1676169   
325   The Lord of the Rings: The Return of the King           1215718   
...                                              ...              ...   
2708                                    The Fighter            275869   
22              The Hobbit: The Desolation of Smaug            483540   
1612                                  Walk the Line            188637   
3237                                           Moon            260607   
874                                           Shrek            467113   

      num_user_for_reviews language  budget  title_year  imdb_score  \
1795                4144.0  English    25.0      1994.0         9.3   
3016                2238.0  English     6.0      1972.0         9.2   
2543                 650.0  English    13.0      1974.0         9.0   
64                  4667.0  English   185.0      2008.0         9.0   
325                 3189.0  English    94.0      2003.0         8.9   
...                    ...      ...     ...         ...         ...   
2708                 389.0  English    25.0      2010.0         7.9   
22                   951.0  English   225.0      2013.0         7.9   
1612                 815.0  English    28.0      2005.0         7.9   
3237                 485.0  English     5.0      2009.0         7.9   
874                  945.0  English    60.0      2001.0         7.9   

      movie_facebook_likes  profit  Rank  
1795                108000    3.34     1  
3016                 43000  128.82     2  
2543                 14000   44.30     3  
64                   37000  348.32     4  
325                  16000  283.02     5  
...                    ...     ...   ...  
2708                 36000   68.57   246  
22                   83000   33.36   247  
1612                 11000   91.52   248  
3237                 47000    0.01   249  
874                      0  207.65   250  

[250 rows x 15 columns]

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [ ]:
# Write your code for creating three new dataframes here
Meryl_Streep = movies[movies['actor_1_name']=='Meryl Streep']
Meryl_Streep.head()

director_name  num_critic_for_reviews   gross  \
392     Nancy Meyers                   187.0  112.70   
1038   Curtis Hanson                    42.0   46.82   
1132     Nora Ephron                   252.0   94.13   
1322   David Frankel                   208.0  124.73   
1390  Robert Redford                   227.0   15.00   

                               genres  actor_1_name             movie_title  \
392              Comedy|Drama|Romance  Meryl Streep       It's Complicated    
1038  Action|Adventure|Crime|Thriller  Meryl Streep         The River Wild    
1132          Biography|Drama|Romance  Meryl Streep          Julie & Julia    
1322             Comedy|Drama|Romance  Meryl Streep  The Devil Wears Prada    
1390               Drama|Thriller|War  Meryl Streep        Lions for Lambs    

      num_voted_users  num_user_for_reviews language  budget  title_year  \
392             69860                 214.0  English    85.0      2009.0   
1038            32544                  69.0  English    45.0      1994.0   
1132            79264                 277.0  English    40.0      2009.0   
1322           286178                 631.0  English    35.0      2006.0   
1390            41170                 298.0  English    35.0      2007.0   

      imdb_score  movie_facebook_likes  profit  
392          6.6                     0   27.70  
1038         6.3                     0    1.82  
1132         7.0                 13000   54.13  
1322         6.8                     0   89.73  
1390         6.2                     0  -20.00

In [ ]:
Leo_Caprio = # Include all movies in which Leo_Caprio is the lead

In [ ]:
Brad_Pitt = # Include all movies in which Brad_Pitt is the lead

In [ ]:
# Write your code for combining the three dataframes here
Combined =

In [ ]:
# Write your code for grouping the combined dataframe here


In [ ]:
# Write the code for finding the mean of critic reviews and audience reviews here


#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

##  Task2 Amazon Prime video data analysis


https://www.kaggle.com/datasets/shivamb/amazon-prime-movies-and-tv-shows?resource=download


Show uniques values of a column 'director'

show all unique values with their counts

get total no of uniwue values of whole data frame

In which year highest no of TV shows and movies were released

how many TV and Movie shows are there in Data frame

show all records with type 'movies; and country united kingdom

show all movie records directed by Paul

Show top 3 Directors, who gave highest no of TV shows and movies released on Prime video

In which year Highest rating show was there

### Task 3 Netflix Analysis
Information about TV shows and Movies
1- upload csv


2- describe, info,dtypes


3- uniques values of each column


4- total no of unique values of Dataframe


5- Unique values with their count


6-is any missing value with count

7- who is the director and show id of show #"ZOO"

8- Convert Datatype of column release date to DateTime

9-In which year highest no of TV shows and Movies relaesed

10-How many movies and TV shows are there in data set

11- Display Titles of all TV shows that were released in " United Sates" only

12- show top 10 Directors who gave highest no of TV shows and Movies on Netflix

13- show the record of all 'Horror' type of Movies

14 What are different 'Ratings' given by Netflix

15- What is Maximum duration of TV show on Netflix

16-sort dataframe by year

